src/box/wal.c

```diff
@@ -782,6 +791,58 @@ wal_set_queue_max_size(int64_t size)
 	journal_queue_set_max_size(size);
 }
 
+/** Retention delay configuration message. */
+struct wal_set_retention_period_msg {
+	/* The state of a synchronous cross-thread call. */
+	struct cbus_call_msg base;
+	/* New retention_period value. */
+	double retention_period;
+};
+
+static int
+wal_set_retention_period_f(struct cbus_call_msg *data)
+{
+	struct wal_writer *writer = &wal_writer_singleton;
+	struct wal_set_retention_period_msg *msg;
+	msg = (struct wal_set_retention_period_msg *)data;
+	xdir_set_retention_period(&writer->wal_dir, msg->retention_period);
+	return 0;
+}
+
+void
+wal_set_retention_period(double period)
+{
+	struct wal_writer *writer = &wal_writer_singleton;
+	struct wal_set_retention_period_msg msg;
+	msg.retention_period = period;
+	cbus_call(&writer->wal_pipe, &writer->tx_prio_pipe,
+		  &msg.base, wal_set_retention_period_f);
+}
+
+static int
+wal_get_retention_vclock_f(struct cbus_call_msg *data)
+{
+	struct wal_writer *writer = &wal_writer_singleton;
+	struct wal_vclock_msg *msg = (struct wal_vclock_msg *)data;
+	xdir_get_retention_vclock(&writer->wal_dir, &msg->vclock);
+	return 0;
+}
+
+void
+wal_get_retention_vclock(struct vclock *vclock)
+{
+	struct wal_writer *writer = &wal_writer_singleton;
+	if (writer->wal_dir.retention_period == 0) {
+		vclock_clear(vclock);
+		return;
+	}
+	struct wal_vclock_msg msg;
+	cbus_call(&writer->wal_pipe, &writer->tx_prio_pipe,
+		  &msg.base, wal_get_retention_vclock_f);
+	if (vclock != NULL)
```

Comment:
Nit: you may assert vclock != NULL in the beginning, or return immediately if vclock == NULL.
Besides you already `vclock_clear`'d the vclock without checking above.

Reply:
Indeed, fixed